boardgame/BoardGameImpl.java

```diff
@@ -0,0 +1,48 @@
+package boardgame;
+
+public class BoardGameImpl implements IBoardGame {
+    private Boolean[][] boardSlot = new Boolean[3][3];
+    // winners are (3,0) (0,3) (3,6) (6,3) (3,3)
+    private int[] sumOfX = {0, 0};
+    private int[] sumOfO = {0, 0};
+    private int xTurnNumber = 1;
+    private int oTurnNumber = 1;
+
+    @Override
+    public String GetWinner(){
+        if(this.xTurnNumber > 3 && this.sumOfX[0] % 3 == 0 && this.sumOfX[1] % 3 == 0){
```

Comment:
ใน if กับ else โค้ดมันเหมือนกันเลย ต่างกันแค่ใช้คนละตัวแปร ดังนั้นเขียน method ในการเช็คออกมาอีกตัวจะลดโค้ดซ้ำและอ่านได้เข้าใจมากขึ้นครับ

var isXWinner = isWinner(this.xTurnNumber, this.sumOfX);
if(isXWinner) return "X";

var isOWinner = isWinner(this.oTurnNumber, this.sumOfO);
if(isOWinner) return "Y";


public bool isWinner(int turnNumber, int[] sumOf) { ... }